commown_self_troubleshooting/static/src/self_troubleshooting_tour.js

```diff
@@ -222,6 +222,132 @@ odoo.define("commown_self_troubleshooting.tour_fp2_battery", function(require) {
+        run: "text J'ai juste besoin d'une vitre de protection",
+      },
+      {
+        content: "Check step 4 nav link is disabled",
+        trigger: "#smartwizard a.nav-link:eq(4).disabled",
+        run: function() {},
+      },
+      {
+        content: "Check step 5 nav link is not disabled",
+        trigger: "#smartwizard a.nav-link:eq(5):not(.disabled)",
+        run: function() {},
+      },
+      commonSteps.gotoNextStep,
+      {
+        content: "Check name",
+        trigger: "input[name=name]",
+        run: function() {
+          const user = $("#top_menu li:last-child a b span:eq(0)").text();
+          if (this.$anchor.val() !== user) {
+            console.log("Field 'name' value should equal the logged-in user name:")
+            console.log("Got: '" + this.$anchor.val() + "' instead of '" + user + "'.");
+            console.log("Error (see above)");
+          }
+        },
+      },
+      commonSteps.gotoNextStep,
+      ...commonSteps.funcAddMoreInfo("text My screen protection need to be replaced!"),
+      ...commonSteps.funcCreateAndCheckTicket("vitre de protection doit"),
+    ]
+  );
+
+  tour.register(
+    "commown_self_troubleshooting_smartphone_need_display_with_protection",
+    { url: "/my" },
+    [
+      {
+        content: "Go to any screen protection page (FP3 in this example)",
+        trigger: 'a[href="/page/self-troubleshoot-fp3-screen"]',
+      },
+      commonSteps.fillInContract,
+      commonSteps.gotoNextStep,
+      {
+        content: "Select yes on the presence of screen protection on this device",
+        trigger: "input[id=has_protection_yes]",
+        run: "text Non",
+      },
+      {
+        content: "Check step 2 nav link is not disabled",
+        trigger: "#smartwizard a.nav-link:eq(2):not(.disabled)",
+        run: function() {},
+      },
+      {
+        content: "Check step 3 nav link is disabled",
+        trigger: "#smartwizard a.nav-link:eq(3).disabled",
+        run: function() {},
+      },
+      commonSteps.gotoNextStep,
+      {
+        content: "Select my display needs to be replaced",
+        trigger: "input[id=replace_screen_yes_step2]",
+        run: "text Mon écran doit être remplacé",
+      },
+      {
+        content: "Check step 4 nav link is not disabled",
+        trigger: "#smartwizard a.nav-link:eq(4):not(.disabled)",
+        run: function() {},
+      },
+      {
+        content: "Check step 5 nav link is not disabled",
+        trigger: "#smartwizard a.nav-link:eq(5):not(.disabled)",
+        run: function() {},
+      },
+      commonSteps.gotoNextStep,
+      {
+        content: "Select I manipulate modules",
+        trigger: "select[id=type_contrat]",
+        run: "text Je manipule les modules en cas de panne",
+      },
+      commonSteps.gotoNextStep,
+      {
+        content: "Check name",
+        trigger: "input[name=name]",
+        run: function() {
+          const user = $("#top_menu li:last-child a b span:eq(0)").text();
```

Comment:
Don't Repeat Yourself (DRY) :smiley: 
Mets ça dans commonSteps avec un nom bien choisi (genre checkInputNameMatchesUser) et réutilise.
Est-ce que ça plante avec un message clair quand la vérification est incorrecte ? Il faut le vérifier.

Reply:
Changé dans le dernier amendement de commit.

+ oui c'est clair dans le log : 

2023-11-06 14:06:44,808 1082 INFO test_12.0_commown_self_troubleshooting odoo.addons.commown_self_troubleshooting.tests.test_tour.TestPageSmartphone: console log: Tour commown_self_troubleshooting_smartphone_need_screen_protection: step button.sw-btn-next succeeded                      
2023-11-06 14:06:45,738 1082 INFO test_12.0_commown_self_troubleshooting odoo.addons.commown_self_troubleshooting.tests.test_tour.TestPageSmartphone: console log: Field 'name' value should equal the logged-in us
er name:                                                                                                                                                                                                          
2023-11-06 14:06:45,742 1082 INFO test_12.0_commown_self_troubleshooting odoo.addons.commown_self_troubleshooting.tests.test_tour.TestPageSmartphone: console log: Got: 'Joel Willis' instead of 'Joel Williscoucou
'.